Sources/ClientRuntime/Networking/Streaming/DataStreamReader.swift

```diff
@@ -42,15 +42,21 @@ public class DataStreamReader: StreamReader {
     }
     
     public func read(maxBytes: UInt? = nil, rewind: Bool = false) -> ByteBuffer {
-        let buffer = ByteBuffer(size: Int(maxBytes ?? availableForRead))
+        var data = Data()
         withLockingClosure {
-            buffer.put(byteBuffer, offset: offset, maxBytes: maxBytes)
-            if !rewind {
-                _availableForRead -= UInt(buffer.length)
-                offset += UInt(buffer.length)
+            let count = Int(maxBytes ?? availableForRead)
+            let ptr = UnsafeMutablePointer<UInt8>.allocate(capacity: count)
+            let buffer = UnsafeMutableBufferPointer(start: ptr, count: count)
+            let bytesRead = byteBuffer.read(buffer: buffer)
+            data = Data(buffer)
+            ptr.deallocate()
```

Comment:
```suggestion
            let ptr = UnsafeMutablePointer<UInt8>.allocate(capacity: count)
            let buffer = UnsafeMutableBufferPointer(start: ptr, count: count)
            defer {
                  buffer.deallocate()
                  data.deallocate()
            }
            let bytesRead = byteBuffer.read(buffer: buffer)
            data = Data(buffer)
```

Reply:
hmm can you explain this change? from my understand, it is only required that we `deallocate` the `UnsafeMutablePointer` (`ptr`).

Reply:
my bad, didn't realize it was BufferPointer, makes sense.